protected/controllers/DisorderController.php

```diff
@@ -86,6 +86,11 @@ public function actionCreate()
         if (isset($_POST['Disorder'])) {
             $disorder->attributes = $_POST['Disorder'];
 
+            if ($disorder->is_ophthalmic === '1' &&
+                (null === $disorder->specialty_id || empty($disorder->specialty_id))) {
```

Comment:
Shouldn't the null be on the right?

Reply:
no

Reply:
I think @apigram is talking about your use of Yoda Conditions https://en.wikipedia.org/wiki/Yoda_conditions